### PR TITLE
[DB] Switch all user_id's value to username field (phone number)

### PR DESCRIPTION
**Pull request by tangziyi001**
Doing this because we want to make sure:
1. Users cannot override each other's info. Since user_id is just a serial number that's easily being stored on multiple devices so they can easily get messed up. On the other hand, username (phone number) is binding to unique user naturally.
2. Adapt to ApiKeyAuthentication where username and apikey are provided together. 

Will change some code on ios too but shouldn't be too much. Since I'm still using user_id in payload but just switch its actual value to username in request/response.

### CODE REVIEW

**Comment by tangziyi001** [comment]
@duckmoll mind taking a look? I think you've also talked about this before.

File: new_dawn_server/actions/api/resources.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def obj_create(self, bundle, **kwargs):
         super(UserActionResource, self).obj_create(bundle).obj.save()
         if bundle.data.get("action_type") == ActionType.LIKE.value:
             NotificationService().send_notification([str(bundle.data.get("user_to_id"))], message="You are liked")
-            if UserAction.objects.filter(user_to_id=bundle.data.get("user_from_id"),
-                                         user_from_id=bundle.data.get("user_to_id"),
+            if UserAction.objects.filter(user_to__username=bundle.data.get("user_from_id"),
+                                         user_from__username=bundle.data.get("user_to_id"),
```
Comment (duckmoll) [inline new_dawn_server/actions/api/resources.py]: This is reading user_to_id from client, but how is it used in username?

Comment (tangziyi001) [reply new_dawn_server/actions/api/resources.py]: In the future, client's notion of "user_id" should all become "username", and we will no longer use the real id anymore

Comment (tangziyi001) [reply new_dawn_server/actions/api/resources.py]: And username will be phone number

Comment (duckmoll) [reply new_dawn_server/actions/api/resources.py]: I see

**Pull request merged**